Lib/urllib/request.py

```diff
@@ -1286,7 +1286,8 @@ def do_open(self, http_class, req, **http_conn_args):
         h.set_debuglevel(self._debuglevel)
 
         headers = dict(req.unredirected_hdrs)
-        headers.update((k, v) for k, v in req.headers.items() if k not in headers)
+        headers.update({k: v for k, v in req.headers.items()
```

Comment:
Please leave as-is.  Don't create an unnecessary intermediate dictionary.

Reply:
I afraid that this change changed semantic.